File: selectSort.js
let unsorted = [14, -4, 17, 6, 22, 1, -5];

function selectSort(arr) {
  /*因為最後一次基本上是最大值了,所以也不用再排,所以是arr.length - 2*/
  for (let j = 0; j <= arr.length - 2; j++) {
    //先找最小值
    let min = Infinity;
    let minIndex = 0;
    //因為跑完一次for loop 代表排好一個元素了,所以i的起始值會是目前執行的次數
    for (let i = j; i <= arr.length - 1; i++) {
      if (arr[i] < min) {
        min = arr[i];
        minIndex = i;
      }
    }
    //目前找到的最小值跟白色的部分的最左邊互換
    let tmp = arr[j];
    arr[j] = arr[minIndex];
    arr[minIndex] = tmp;
  }
  console.log(arr);
  return arr;
}

selectSort(unsorted);

File: selectSort.py
#selectSort =>分為已排序(左)跟未排序(右) , 在未排序的array中選擇最小的, 換到已排序的最後面

arr = [15,669,45,51,848,6,21,48,954,877,32,54,65,3345,31,248,35,8,9999]

def selectSort(arr:list)->list:
    for i in range(0,len(arr)-1,1):
        min = float('inf')
        idx = None
        for j in range(i,len(arr),1):
            if arr[j] < min:
                min = arr[j]
                idx = j
        arr[i],arr[idx] = arr[idx],arr[i]


if __name__ == '__main__':
    selectSort(arr)
    print(arr)